core/common/src/main/java/alluxio/security/authorization/AccessControlList.java

```diff
@@ -237,7 +247,13 @@ public void setEntry(AclEntry entry) {
     // TODO(cc): when setting non-mask entries, the mask should be dynamically updated too.
     switch (entry.getType()) {
       case NAMED_USER:  // fall through
-      case NAMED_GROUP: // fall through
+      case NAMED_GROUP:
+        if (mExtendedEntries == null) {
+          mExtendedEntries = new ExtendedACLEntries();
+        }
+        mExtendedEntries.setEntry(entry);
+        updateMask();
+        return;
       case MASK:
```

Comment:
Does setting the mask directly need to compute anything additional?

Reply:
nope.